src/views/Update/components/UpdatePage/index.js

```diff
@@ -0,0 +1,365 @@
+   * @returns {Array<Object>} Array of updated fields in a subobject
+   */
+  function getUpdatedFields(updatedInfo, currentInfo) {
+    var updatedFields = [];
+    for (const field in currentInfo) {
+      if (updatedInfo[field] !== currentInfo[field]) {
+        updatedFields.push({
+          field: field,
+          value: currentInfo[field],
+          label: getAssociatedLabel(field),
+        });
+      }
+    }
+    return updatedFields;
+  }
+
+  //handles confirmation window's final submit
+  const handleConfirm = () => {
+    setSaving(true);
+    let updateRequest = getUpdatedFields(currentInfo, updatedInfo);
+    let updateRequestEmail = getUpdatedFields(currentInfo, updatedInfo);
+    updateRequestEmail.push({
+      field: 'changeReason',
+      value: changeReason,
+      label: 'Reason for change',
+    });
+    requestInfoUpdate(updateRequestEmail, updateRequest).then(() => {
+      createSnackbar(
+        'A request to update your information has been sent. Please check back later.',
+        'info',
+      );
+      setSaving(false);
+      handleWindowClose();
+    });
+  };
+
+  //executes on confirmation window cancel & submit
+  const handleWindowClose = () => {
+    setOpenConfirmWindow(false);
+    setChangeReason('');
+    setConfirmRows('');
+  };
+
+  const handleSaveButtonClick = () => {
+    if (updatedInfo.firstName === '' || updatedInfo.lastName === '') {
+      createSnackbar('Please fill in your first and last name.', 'error');
+    } else {
+      getCurrentChanges(currentInfo, updatedInfo);
+      setOpenConfirmWindow(true);
+    }
+  };
+
+  const saveButton = isSaving ? (
+    <GordonLoader size={32} />
+  ) : (
+    <Button
+      variant="contained"
+      color="secondary"
+      onClick={handleSaveButtonClick}
+      disabled={hasNoChange}
+    >
+      Update
+    </Button>
+  );
+
+  /**
+   * @param {Array<Object>} fields static fields that define label, name, value, type
+   * @returns JSX elements of all elements in array with correct type
+   */
+  const infoMap = (fields) => {
+    return fields.map((field) => (
+      <ProfileUpdateField
+        label={field.label}
+        name={field.name}
+        value={updatedInfo[field.name]}
+        type={field.type}
+        menuItems={field.menuItems}
+        onChange={handleChange}
+      />
+    ));
+  };
+
+  // following 3 consts determine confirmation window and its contents
+  const [confirmRows, setConfirmRows] = useState('');
+  const [changeReason, setChangeReason] = useState('');
+  const getCurrentChanges = (currentInfo, updatedInfo) => {
+    setConfirmRows(
+      <Grid
+        container
+        direction="row"
+        style={{
+          width: '100%',
+          minWidth: 504,
+        }}
+      >
+        {getUpdatedFields(currentInfo, updatedInfo).map((field) => (
+          <ConfirmationRow field={field} prevValue={currentInfo[field.field]} />
+        ))}
+      </Grid>,
+    );
```

Comment:
There is no reason to have `confirmRows` as a state variable. Just put this JSX inline where `{confirmRows}` is currently used. Making it a state variable just complicates the code and introduces more room for bugs (especially state update race conditions).